[WIP] Terraform Docs for TF policy (#2690)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/write/tests.mdx

---
page_title: Write and run policy tests
description: >-
  Learn how to write and run Terraform policy tests locally to ensure they work as expected.
---

# Write and run policy tests

Once you have authored a policy, write tests that verify that your policy works as intended. Then, run your tests before you implement your new policy and whenever you update your policy.

@include "partials/beta.mdx"

## Author policy tests

> Each test file will include a set of mock providers, resources, data sources, and modules that Terraform policy will use to test your policies. Policy test files must end with `.policytest.hcl`. We recommend that you include your tests in the same VCS repository as your policies. Refer to [Write policies - Directory structure](/terraform/policy/write#directory-structure) for more information about organizing your policies and tests.

Reviewer: ```suggestion
Each test file includes a set of mock providers, resources, data sources, and modules that Terraform policy uses to test your policies. Policy test files must have the `.policytest.hcl` extension. 

We recommend storing your tests in the same VCS repository as your policies. Refer to [Directory structure](/terraform/policy/write#directory-structure) for more information about organizing your policies and tests.
```

To learn how to write policy tests, refer to the [Policy test reference documentation](/terraform/policy/reference/test).

## Run tests

The `tfpolicy test` command runs your tests locally before you enforce them against your infrastructure, and whenever you update your policies.

> ### Overview

Reviewer: ```suggestion
Run the `tfpolicy test` command to execute your tests locally before enforcing them against your infrastructure. You should also test policies after any updates.

Specify the path to the policies you want to test with the `--policies` flag: 
```

```shell-session
$ tfpolicy test --policies=/path/to/policies
```

> You can provide separate paths to the policy files and the test files. If you only specify the path to your policy files, the tests in that path will be executed.

Reviewer: ```suggestion
You can provide separate paths to the policy files and the test files. If you only specify the path to your policy files, Terraform policy only executes tests in that path.
```

```shell-session
$ tfpolicy test --policies=/path/to/policies --tests=/path/to/tests
```

> When it runs your tests, Terraform policy will load values for input variables from your local environment, with the prefix `TF_VAR_`. For example:

Reviewer 2: `TF_VAR_` prefix is not supported. Refer the ways to provide inputs to test in the thread https://ibm-hashicorp.slack.com/archives/C09KHSUK0QP/p1783407006989349?thread_ts=1783366683.420259&cid=C09KHSUK0QP
cc: @vpaul97 @viveksinghgit95

Reviewer 3: tf_var_ is for terraform variables, here command we are running is cli , so currently for cli input these are the ways to pass the values 
1.per test case, with an inputs { bucket_name = "case-level" } block inside a resource/provider/module block
2.for a whole test file, with a top-level inputs { bucket_name = "file-level" }
3.on the command line with --input bucket_name=my-bucket
4. via the env var TFPOLICY_INPUT_bucket_name=my-bucket
5. from a var file, --input-file=vars.policyvars.hcl containing bucket_name = "my-bucket"

> When it runs your tests, Terraform policy will load values for input variables from your local environment, with the prefix `TF_VAR_`. For example:

Reviewer: ```suggestion
When it runs your tests, Terraform policy loads values for input variables from your local environment that match the `TF_VAR_` prefix. For example:
```
